File: src-py/use_cases/get_service_order_by_id.py
from typing import Optional

from schemas.preventive import GetServiceOrderByIdParams, ServiceOrder
from utils.prisma import prisma


async def getServiceOrderById(params: dict) -> Optional[ServiceOrder]:
    validated = GetServiceOrderByIdParams.model_validate(params)

    service_order = await prisma.preventiveos.find_unique(
        where={"id": validated.id},
        include={
            "nature": True,
            "machine": True,
            "responsible": True,
            "actions": {"include": {"nature": True, "machine": True}},
            "actionsTaken": {
                "include": {
                    "action": {"include": {"nature": True, "machine": True}},
                }
            },
        },
    )

    if service_order is None:
        return None

    return ServiceOrder.model_validate(service_order.model_dump())

File: server/use-cases/getServiceOrderById.ts
import prisma from "../utils/prisma"

import {
    GetServiceOrderByIdParamsType,
} from '../../schemas/preventive'

export async function getServiceOrderById({ id }: GetServiceOrderByIdParamsType) {
    try {
        const serviceOrder = await prisma.preventiveOS.findUnique({
            where: { id },
            include: {
                nature: true,
                machine: true,
                responsible: true,
                actions: {
                    include: {
                        nature: true, machine: true
                    }
                },
                actionsTaken: {
                    include: {
                        action: {
                            include: {
                                nature: true, machine: true
                            }
                        },
                    }
                }
            }
        })
        return serviceOrder
    } catch (error) {
        throw error
    }
}
